src/placeos-frontend-loader/api/remote/remote.cr

```diff
@@ -65,8 +81,25 @@ module PlaceOS::FrontendLoader
       getter hash : String
       getter tag : String | Nil
 
-      def initialize(@repository : Model::Repository, @branch : String? = "master", @tag : String? = nil)
+      def initialize(url : String | URI, @branch : String? = "master", @tag : String? = nil, @hash : String? = "HEAD")
+        uri = url.is_a?(URI) ? url : URI.parse(url)
+        @repo_name = uri.path.strip("/")
+        @remote_type = {% begin %}
+          case uri.host.to_s
+            {% for remote in Reference::Type.constants %}
+          when .includes?(Reference::Type::{{ remote }}.to_s.downcase)
+            Reference::Type::{{ remote.id }}
+          {% end %}
+          else
+            raise Exception.new("Host not supported: #{url}")
+          end
+          {% end %}
+        @repository = Model::Repository.new(name: @repo_name, folder_name: @repo_name, uri: uri, commit_hash: @hash, branch: branch)
```

Comment:
I don't think there's any need for this model to be a part of the `Reference` struct